src/actions/index.js

```diff
@@ -0,0 +1,26 @@
+import sampleData from './sampleData';
+
+/**
+ * Makes the mock "API request"
+ */
+const fetchProducts = () => {
+  return new Promise((resolve) => {
+    window.setTimeout(() => {
+      resolve(sampleData.products);
+    }, 1000);
+  });
+}
+
+/**
+ * Retrieves product list from mock API
+ */
+export const fetchProductsFromAPI = () => {
+  return (dispatch) => {
+    return fetchProducts().then(
+      (resp) => dispatch({
```

Comment:
In this case, it's nice to just name the parameter `products` and take advantage of [object property shorthand](https://ariya.io/2013/02/es6-and-object-literal-property-value-shorthand):

```javascript
return fetchProducts().then(products => {
  dispatch({
    type: 'UPDATE_PRODUCTS',
    products,
  });
});
```